interop/xds_federation/client.go

```diff
@@ -0,0 +1,123 @@
+	soakMinTimeMsBetweenRPCs               = flag.Int("soak_min_time_ms_between_rpcs", 0, "The minimum time in milliseconds between consecutive RPCs in a soak test (rpc_soak or channel_soak), useful for limiting QPS")
+	testCase                               = flag.String("test_case", "rpc_soak",
+		`Configure different test cases. Valid options are:
+        rpc_soak: sends --soak_iterations large_unary rpcs;
+        channel_soak: sends --soak_iterations rpc, rebuilding the channel each time`)
+
+	logger = grpclog.Component("interop")
+)
+
+type credsMode uint8
+
+const (
+	credsNone credsMode = iota
+	credsComputeEngineCreds
+)
+
+type clientConfig struct {
+	tc   testgrpc.TestServiceClient
+	opts []grpc.DialOption
+	uri  string
+}
+
+func main() {
+	flag.Parse()
+	logger.Infof("Client running with test case %q", *testCase)
+	var clients []clientConfig
+	uris := strings.Split(*serverUris, ",")
+	creds := strings.Split(*credentialsTypes, ",")
+	if len(uris) != len(creds) {
+		logger.Fatalf("Number of entries in --server_uris (%d) != number of entries in --credentials_types (%d)", len(uris), len(creds))
+	}
+	for i := range uris {
+		var opts []grpc.DialOption
+		if creds[i] == computeEngineCredsName {
+			opts = append(opts, grpc.WithCredentialsBundle(google.NewComputeEngineCredentials()))
+		} else if creds[i] == insecureCredsName {
+			opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		} else {
+			logger.Fatalf("Unsupported credentials type: %v", creds[i])
+		}
+		cc, err := grpc.Dial(uris[i], opts...)
+		if err != nil {
+			logger.Fatalf("Fail to dial %v: %v", uris[i], err)
+		}
+		defer cc.Close()
+		clients = append(clients, clientConfig{
+			tc:   testgrpc.NewTestServiceClient(cc),
+			opts: opts,
+			uri:  uris[i],
+		})
+	}
+	var wg sync.WaitGroup
+	for i := range clients {
+		wg.Add(1)
+		go func(c clientConfig) {
+			defer wg.Done()
+			switch *testCase {
+			case "rpc_soak":
+				interop.DoSoakTest(c.tc, c.uri, c.opts, false /* resetChannel */, *soakIterations, *soakMaxFailures, time.Duration(*soakPerIterationMaxAcceptableLatencyMs)*time.Millisecond, time.Duration(*soakMinTimeMsBetweenRPCs)*time.Millisecond, time.Now().Add(time.Duration(*soakOverallTimeoutSeconds)*time.Second))
+				logger.Infof("RpcSoak done for server: %s", c.uri)
+			case "channel_soak":
+				interop.DoSoakTest(c.tc, c.uri, c.opts, true /* resetChannel */, *soakIterations, *soakMaxFailures, time.Duration(*soakPerIterationMaxAcceptableLatencyMs)*time.Millisecond, time.Duration(*soakMinTimeMsBetweenRPCs)*time.Millisecond, time.Now().Add(time.Duration(*soakOverallTimeoutSeconds)*time.Second))
+				logger.Infof("ChannelSoak done for server: %s", c.uri)
+			default:
+				logger.Fatal("Unsupported test case: ", *testCase)
```

Comment:
Can this be checked up top along with the check for `len(uris) != len(creds)`?

Reply:
We could do that, but then we'd need to create an enum type to complete the switch without double-checking, which does not feel worth it for this code.

PLMK your thoughts, though.

Reply:
My thoughts were that we could validate up top (for all flags), that they contain only supported values, and exit if necessary. Once we do that, we are guaranteed in the rest of the code that flags only contain supported values, and we don't need to have `default` cases in switch statements.

Reply:
sg, done